src/compiler/utilities.ts

```diff
@@ -1471,14 +1471,75 @@ namespace ts {
         return getSourceTextOfNodeFromSourceFile(sourceFile, str).charCodeAt(0) === CharacterCodes.doubleQuote;
     }
 
+    export function getJSInitializerSymbol(symbol: Symbol) {
+        if (!symbol || !symbol.valueDeclaration) {
+            return symbol;
+        }
+        const declaration = symbol.valueDeclaration;
+        const e = getDeclaredJavascriptInitializer(declaration) || getAssignedJavascriptInitializer(declaration);
+        return e ? e.symbol : symbol;
+    }
+
+    export function getDeclaredJavascriptInitializer(node: Node) {
+        if (node && isVariableDeclaration(node) && node.initializer) {
+            return getJavascriptInitializer(node.initializer) ||
+                isIdentifier(node.name) && getDefaultedJavascriptInitializer(node.name, node.initializer);
+        }
+    }
+
+    export function getAssignedJavascriptInitializer(node: Node) {

```

Comment:
where else do we need the BarBartoken handling that is not in getDefaultedJavascriptInitializer? can we consolidate this in one place.